velox/functions/prestosql/ArithmeticImpl.h

```diff
@@ -44,10 +44,13 @@ round(const TNum& number, const TDecimals& decimals = 0) {
   }
 
   double factor = std::pow(10, decimals);
+  static const double kInf = std::numeric_limits<double>::infinity();
   if (number < 0) {
-    return (std::round(number * factor * -1) / factor) * -1;
+    return (std::round(std::nextafter(number, number - kInf) * factor * -1) /
```

Comment:
I'm confused by number - kInf and number + kInf. Aren't these just -kInf and kInf?

Reply:
Also, if second arg to nextafter is double, wouldn't the result be double as well? I assume we want to get float result if 'number' is float, no?

Reply:
Thanks a lot. I added float support in this function, and I added corresponding test. Float is different with double. I need define kInf as float type to make nextafter function work.